### PR TITLE
Read CSV file

### PR DESCRIPTION
**Pull request by evertsd**
Added support for parsing a CSV file of pokemon

Seems to work well, one thing that definitely still needs to be done here is moving classes into their own files / whatever refactoring comes with that (hopefully minimal)

I know you're still actively working on this project, but thought I'd share the code in-case you're interested. 

Thanks for the base project, <3 being able to sort by max / min IV when uploading the CSV file to a  google spreadsheet.

### CODE REVIEW

**Comment by goatslacker** [comment]
This is pretty cool.

Suggestion: can we make this a different script? `parseCSV` or whatever? I'm trying to break things out of `index.js`.

More comments coming

File: index.js [hunk 1]
```diff
@@ -1,9 +1,133 @@
+const parse = require('./node_modules/csv-parser');
```
Comment (goatslacker) [inline index.js]: You can just `const csvParser = require('csv-parser');`

**Comment by goatslacker** [comment]
Scratch that ^ maybe keeping this functionality in index.js is not a bad idea. The code should still live in a separate file though.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (goatslacker) [inline package.json:1]: Thanks :) I was meaning to do this.

File: src/magic.js [hunk 3]
```diff
@@ -8,69 +8,118 @@ const logPokemon = require('./logPokemon')
...
+  this.maybe = this.maybeValues.length > 0
+  this.valuesRange = this.findValuesRange(results);
+}
+
+
+IvResults.prototype.isValid = function() {
+  return !this.errors.length;
+}
+
+IvResults.prototype.toString = function() {
```
Comment (goatslacker) [inline src/magic.js]: yay!

File: src/magic.js [hunk 4]
```diff
@@ -8,69 +8,118 @@ const logPokemon = require('./logPokemon')
...
-//  const sortedByBest = values.sort((a, b) => {
-//    return a.percent.PerfectIV > b.percent.PerfectIV ? -1 : 1
-//  })
+  return results;
+}
 
-  // assume that we caught this pokemon in the wild where there are no
-  // half levels
-//  const mostProbable = sortedByBest.filter(x => x.Level % 1 === 0)
+const IvResults = function(pokemon, results) {
```
Comment (goatslacker) [inline src/magic.js]: this looks like a class so it should be a class

**Comment by evertsd** [comment]
Totally agree with the comments thus far. I'm fine with making it a different script, why'd you change your mind?

**Comment by evertsd** [comment]
Ok I think I remedied all those issues, and still seems to work just as well

**Comment by goatslacker** [comment]
Sweet!

**Pull request merged**